RecoLocalTracker/SiPixelClusterizer/plugins/PixelThresholdClusterizer.cc

```diff
@@ -119,6 +144,11 @@ void PixelThresholdClusterizer::clusterizeDetUnitT( const T & input,
   
   detid_ = input.detId();
   
+  // Set separate cluster threshold for L1 (needed for phase1)
+  auto clusterThreshold = theClusterThreshold;
+  int layer = (DetId(detid_).subdetId()==1) ? PXBDetId(detid_).layer() : 0;
```

Comment:
@venturia 
This class was already using PXBDetId to get the layer number elsewhere. TTopo gives the same value for layer. It indeed could be changed to use TrackerTopology, but EventSetup is not easily accessible in this class. The class which inherits from this could pass it I think.